### PR TITLE
Don't allow metrics to be stored in any case

### LINKED ISSUE #2743: Avoid writing event.metrics to etcd (only store record).
## Expected Behavior
As a user of the sensu-prometheus-collector check, I would like the check to only result in metrics events, so that the metrics are not stored in Sensu.

## Current Behavior
All checks are stored along with metrics data in Sensu.

## Possible Solution 1
Add a flag called "metrics_only" to check configurations, so that successful check executions are only transmitted as metrics events, and thus not stored. Unsuccessful checks are still transmitted as regular check events, so that users can easily detect failure.

## Possible Solution 2
Add a flag called "no_store_metrics" to check configurations, so that extracted metrics are not stored along with the check event.

## Context
A user is attempting to push a large amount of metrics through Sensu, but the default etcd request size limit is hampering that effort.

### PR DESCRIPTION
**Pull request by echlebek**
## What is this change?

This commit explicitly nils out metrics when events arrive at the
store, in order to ensure that metrics are never stored in etcd.

Care is taken to not write to the pointer that is given to the
store. To this end, some unrelated existing code in the same
UpdateEvent method has also been modified to be safer.

## Why is this change necessary?

Closes #2743 

## Does your change need a Changelog entry?

Included

## Have you reviewed and updated the documentation for this change? Is new documentation required?

This change brings us in line with the documentation.

## How did you verify this change?

A new integration test.

### CODE REVIEW

File: backend/store/etcd/event_store_test.go [hunk 1]
```diff
@@ -112,3 +112,24 @@ func TestEventStorage(t *testing.T) {
 		assert.Error(t, err)
 	})
 }
+
+func TestDinnaeStoreMetrics(t *testing.T) {
```
Comment (nikkiki) [inline backend/store/etcd/event_store_test.go]: I don't think I'm smart enough to know what `Dinnae` means..

Comment (echlebek) [reply backend/store/etcd/event_store_test.go]: https://en.wikipedia.org/wiki/Scotticism

Comment (ccressent) [reply backend/store/etcd/event_store_test.go]: I had to consult that wiki page too :(
Should it just read `TestDoNotStoreMetrics()`?

File: backend/store/etcd/event_store.go [hunk 2]
```diff
@@ -161,7 +161,7 @@ func (s *Store) GetEventByEntityCheck(ctx context.Context, entityName, checkName
 
 // UpdateEvent updates an event.
 func (s *Store) UpdateEvent(ctx context.Context, event *types.Event) error {
-	if event.Check == nil {
+	if event == nil || event.Check == nil {
 		return errors.New("event has no check")
```
Comment (ccressent) [inline backend/store/etcd/event_store.go]: Is that the correct error for the `event == nil` case? I think it should be something along the lines of "cannot update nil event" or something.

Comment (echlebek) [reply backend/store/etcd/event_store.go]: The == nil case is really just a guard against a panic, not sure it deserves its own error case.

**Pull request merged**